### PR TITLE
New recipe for mauveAligner

### PR DESCRIPTION
**Pull request by jamesabbott**
* [*] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [*] This PR adds a new recipe.
* [*] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

New PR to replace #9498 which had become entangled with other recipes...

This is a component of Mauve, providing the mauveAligner and progressiveMauve tools

Mauve is a rather complex build which I have working (manually) under linux, however Clang++ does not like some definitions so osx builds are disabled for now. I'll try to sort these out once the whole linux build is working

### CODE REVIEW

File: recipes/mauvealigner/meta.yaml [hunk 1]
```diff
@@ -38,3 +38,4 @@ about:
   license_family: GPL
   summary: The mauveAligner and progressiveMauve command-line tools for generating multiple genome alignments in the presence of large-scale evolutionary events
   dev_url: https://sourceforge.net/projects/mauve/
+
```
Comment (mbargull) [inline recipes/mauvealigner/meta.yaml]: You can also create empty commits using the `--allow-empty` argument, e.g., via
```
git commit --allow-empty -m 'trigger build'
```

Comment (jamesabbott) [reply recipes/mauvealigner/meta.yaml]: Neat...every day's a school day...

**Comment by mbargull** [comment]
Thanks, LGTM. I just removed the `boost` dependency, hoping that `boost-cpp` covers all that is needed. That way the package's dependency might be a little more lightweight since `python` isn't pulled in by `boost`.
If that builds fine, feel free to merge!

**Comment by jamesabbott** [comment]
All looks good - thanks

**Pull request merged**